source/common/router/config_impl.cc

```diff
@@ -379,6 +379,15 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHostImpl& vhost,
       throw EnvoyException(absl::StrCat("Duplicate upgrade ", upgrade_config.upgrade_type()));
     }
   }
+
+  if (route.route().has_regex_rewrite()) {
+    if (!prefix_rewrite_.empty()) {
+      throw EnvoyException("Cannot specify both prefix_rewrite and regex_rewrite");
```

Comment:
Please a test case for this. You can use EXPECT_THROW_WITH_MESSAGE for testing.

Reply:
I've added this test case in the latest commit.